addon/components/dialog.js

```diff
@@ -0,0 +1,91 @@
+import Component from '@glimmer/component';
+import { typeOf } from '@ember/utils';
+import { guidFor } from '@ember/object/internals';
+
+import { modifier } from 'ember-modifier';
+
+import { Keys } from 'ember-headlessui/utils/keyboard';
+
+export default class DialogComponent extends Component {
+  DEFAULT_TAG_NAME = 'div';
+
+  guid = `${guidFor(this)}-headlessui-dialog`;
+  $portalRoot = document.body;
```

Comment:
Another option would be to target the `.ember-application` element; for the app running normally, this will be the body anyway, but for testing it's down inside the preview window.

Reply:
Nice idea.